### ISSUE TITLE
Running chalice locally and executing browser request produces route error

### ISSUE THREAD

**Issue by olingern** [aws/chalice#366]
Basic Chalice setup:
```python
from chalice import Chalice
app = Chalice(app_name='hello_world')

@app.route('/')
def index():

app = Chalice(app_name='hello_world')

@app.route('/')
def index():
    return {'hello': 'world' }
```

Run on local port:
`chalice local --port=8001`

Make a request (GET) using any browser

Error:
<img width="1070" alt="screen shot 2017-06-04 at 5 01 33 pm" src="https://cloud.githubusercontent.com/assets/1470297/26765314/8a668970-4947-11e7-8f9f-9884daf1453a.png">

Due to browsers looking for a favicon, the server should handle this by returning early and not attempt to search for a valid route.  Adding an early return [prior to this loop](https://github.com/awslabs/chalice/blob/master/chalice/local.py#L56) based upon a config dictionary would seem to be a good fix.

Happy to submit a PR for this, let me know where the config vars for something like `ignored_routes` should go.

Cheers

**Comment by stealthycoin** [comment]
This seems more like an enhancement to make our local fake version of API gateway respond the same way that the real API gateway would.

**Comment by stealthycoin** [comment]
It looks like I missed linking this issue as this was resolved when chalice local was reworked here #540. It now returns a 403 for /favicon.ico which matches what the real API Gateway does.
